app/questionnaire/router.py

```diff
@@ -270,15 +271,31 @@ def _is_section_enabled(self, section):
         if "enabled" not in section:
             return True
 
-        for condition in section["enabled"]:
-            if evaluate_when_rules(
-                condition["when"],
+        enabled = section["enabled"]
+        if isinstance(enabled, Sequence):
+            for condition in enabled:
+                if evaluate_when_rules(
+                    condition["when"],
+                    self._schema,
+                    self._metadata,
+                    self._answer_store,
+                    self._list_store,
+                ):
+                    return True
+        else:
+            location = Location(section_id=section["id"])
+            when_rule_evaluator = WhenRuleEvaluator(
                 self._schema,
-                self._metadata,
                 self._answer_store,
                 self._list_store,
-            ):
-                return True
+                self._metadata,
+                self._response_metadata,
+                location=location,
+                routing_path_block_ids=None,
+            )
+
+            return when_rule_evaluator.evaluate(enabled.get("when"))
```

Comment:
The `when` key has to exist so don't need `.get()`.

Reply:
Removed unnecessary `.get()`, thanks!